source/features/deep-blame.tsx

```diff
@@ -0,0 +1,111 @@
+import './deep-blame.css';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import React from 'dom-chef';
+import features from '../libs/features';
+import {getRepoGQL, getReference} from '../libs/utils';
+import delegate, {DelegateEvent} from 'delegate-it';
+import versionIcon from 'octicon/versions.svg';
+import octofaceIcon from 'octicon/octoface.svg';
+
+const getPullRequestBlameCommit = async (commit: string): Promise<string | undefined> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "${commit}") {
+				... on Commit {
+					associatedPullRequests(last: 1) {
+						nodes {
+							mergeCommit {
+								oid
+							}
+							commits(last: 1) {
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const {nodes} = repository.object.associatedPullRequests;
+	const mergeCommit = nodes[0].mergeCommit.oid;
+
+	if (mergeCommit === commit) {
+		return nodes[0].commits.nodes[0].commit.oid;
+	}
+
+	return undefined;
+};
+
+function getCommitHash(commit: HTMLElement): string {
+	return (commit.nextElementSibling! as HTMLAnchorElement).href.split('/').slice(-1)[0];
+}
+
+async function redirectToBlameCommit(event: DelegateEvent<MouseEvent, HTMLLinkElement>): Promise<void | false> {
+	const {lineNumber, commit: pullRequestCommit} = event.delegateTarget.dataset;
+	if (event.delegateTarget.href && event.altKey) {
+		event.preventDefault();
+	}
+
+	event.delegateTarget.classList.add('anim-pulse');
+	event.delegateTarget.replaceChild(octofaceIcon(), event.delegateTarget.firstElementChild!);
+
+	const prBlameCommit = await getPullRequestBlameCommit(String(pullRequestCommit));
+	if (!prBlameCommit) {
+		event.delegateTarget.removeChild(event.delegateTarget.firstElementChild!);
+		return;
+	}
+
+	const href = new URL(window.location.href.replace(String(getReference()), prBlameCommit));
+	href.hash = 'L' + String(lineNumber);
+	location.href = String(href);
```

Comment:
Are other links loaded via AJAX? Can we trigger an AJAX navigation? You could create a util function called `navigateTo`

Reply:
Where do you have an example that I should follow.
This is the only place we navigate

Reply:
I tested and other links are not loaded via AJAX, so we're good here